sdks/python/apache_beam/io/gcp/bigquery_file_loads.py

```diff
@@ -387,6 +387,14 @@ def process(self, element, load_job_name_prefix, *schema_side_inputs):
     else:
       schema = self.schema
 
+    import unicode
+    import json
+
+    if isinstance(schema, (str, unicode)):
+      schema = bigquery_tools.parse_table_schema_from_json(schema)
+    elif isinstance(schema, dict):
```

Comment:
Could you explain the underlying issue here ? (couldn't find details in the JIRA).

Could the issue be that you are specifying an incompatible/incorrect schema object ? Can you try specifying schema as a TableSchema object ? For example, see here: https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/gcp/bigquery_write_it_test.py#L80

If you are fixing an actual underlying issue here, please add a unit test.

Reply:
Apologies if the issue as described was not clear. 

Basically - I was trying to creating a dataflow template, which requires using ValueProvider argument. I do not believe it is possible to pass a serialized object as the argument in this case.

I'm actually borrowing the logic from here: https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/gcp/bigquery.py#L757

Reply:
Thanks. Can you please add a unit test.

Reply:
@chamikaramj I am working on it. Having a very difficult time trying to get the tests running locally.

Reply:
Thanks. This might help: https://cwiki.apache.org/confluence/display/BEAM/Python+Tips